src/transformers/api-history.ts

```diff
@@ -0,0 +1,189 @@
+    throw new Error('GH_TOKEN is required when running in GitHub Actions.');
+  }
+
+  // TODO: Remove this
+  if (process.env.LOCAL_DEV === 'true') {
+    const versions: PrReleaseVersionsContainer = {
+      35658: {
+        release: 'v30.0.0-nightly.20231214',
+        backports: ['v29.0.0-alpha.9'],
+      },
+      41391: {
+        release: 'v31.0.0-alpha.1',
+        backports: [] as string[],
+      },
+      42086: {
+        release: 'v32.0.0-nightly.20240531',
+        backports: ['v31.0.0'],
+      },
+    };
+
+    _allPrReleaseVersions = versions;
+    return _allPrReleaseVersions;
+  }
+
+  // ? Maybe log this?
+  if (!process.env.GH_TOKEN) {
+    _allPrReleaseVersions = {};
+    return _allPrReleaseVersions;
+  }
+
+  const fetchOptions = {
+    method: 'GET',
+    headers: {
+      Accept: 'application/vnd.github+json',
+      'X-GitHub-Api-Version': '2022-11-28',
+      Authorization: `Bearer ${process.env.GH_TOKEN}`,
+    },
+  };
+
+  const artifactsListResponse = await fetch(
+    'https://api.github.com/repos/electron/website/actions/artifacts',
+    fetchOptions
+  );
+  const latestArtifact = (await artifactsListResponse.json()).artifacts
+    .filter(({ name }) => name === 'resolved-pr-versions')
+    .sort((a: { id: number }, b: { id: number }) => a.id > b.id)[0];
+
+  // ? Maybe use streams/workers
+  const archiveDownloadResponse = await fetch(
+    latestArtifact.archive_download_url,
+    fetchOptions
+  );
+  const buffer = Buffer.from(await archiveDownloadResponse.arrayBuffer());
+
+  const zip = new AdmZip(buffer);
+  const parsedData = JSON.parse(
+    zip.readAsText(zip.getEntries()[0])
+  ) as PrReleaseArtifact;
+
+  if (!parsedData?.data) {
+    throw new Error('No data found in the artifact.');
+  }
+
+  _allPrReleaseVersions = parsedData.data;
+  return _allPrReleaseVersions;
+}
+
+// Most of this is copy-pasted from: <https://github.com/electron/website/blob/ac3bab3131fc0f5de563574189ad5eab956a60b9/src/transformers/js-code-blocks.ts>
+async function transformer(tree: Parent) {
+  let needImport = false;
+  // TODO: Filter out unreleased versions using @electron/fiddle-core
+  const allPrReleaseVersions = await getAllPrReleaseVersions();
+  visitParents(tree, matchApiHistoryCodeBlock, maybeGenerateApiHistoryTable);
+  visitParents(tree, 'mdxjsEsm', checkForApiHistoryTableImport);
+
+  if (needImport) {
+    tree.children.unshift(getJSXImport('ApiHistoryTable'));
+  }
+
+  function checkForApiHistoryTableImport(node: Node) {
+    if (
+      isImport(node) &&
+      node.value.includes('@site/src/components/ApiHistoryTable')
+    ) {
+      needImport = false;
+    }
+  }
+
+  function maybeGenerateApiHistoryTable(
+    node: Code,
+    ancestors: Parent[]
+  ): ActionTuple | void {
+    const parent = ancestors[0];
+    const idx = parent.children.indexOf(node);
+
+    const apiHistory = parseYaml(node.value) as ApiHistory;
+
+    const versionsInHistory = [];
```

Comment:
```suggestion
    const prsInHistory = [];
```

Current name is a bit inaccurate.

Reply:
Resolved in [`c927a72` (#594)](https://github.com/electron/website/pull/594/commits/c927a722b773b90c355da10f0d0519b0697469c2)